sqlglot/dialects/hive.py

```diff
@@ -488,7 +490,7 @@ def datatype_sql(self, expression: exp.DataType) -> str:
                 expression = exp.DataType.build("text")
             elif expression.this in exp.DataType.TEMPORAL_TYPES:
                 expression = exp.DataType.build(expression.this)
-            elif expression.is_type("float"):
+            elif expression.is_type("float", only_kind=True):
```

Comment:
why are you adding only_kind everywhere? float should basically always equal float right?

array == array

array[int] != array[str]

Reply:
> float should basically always equal float right?

my guess to this is basically the same as in https://github.com/tobymao/sqlglot/pull/2096#discussion_r1299437320: the reason i implemented it like this was to ensure we don't assume any kind of equality between types that don't share the same precision